src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -108,22 +117,60 @@ class RedownloadService[F[_]: NonEmptyParallel](
     HeightRange(minHeight(s), maxHeight(s))
   }
 
-  def getMajorityGapRanges: F[List[HeightRange]] = lastMajorityState.get.map(missingProposalFinder.findGapRanges)
-
   def replacePeerProposals(peer: Id, proposals: SnapshotProposalsAtHeight): F[Unit] =
-    peersProposals(peer).set(proposals.some)
+    peerProposals(peer).set(proposals.some)
 
-  def persistPeerProposal(peer: Id, proposal: Signed[SnapshotProposal]): F[Unit] =
+  def removePeerProposalsBelowHeight(height: Long): F[Unit] =
```

Comment:
I'm not sure if we should just remove proposals based on our calculated majority. In case someone else gets stuck because of a missing proposal, we may not have that proposal anymore at the time when the node will be asking for it. I would take into account the filter and check if all other nodes have that proposal/majority picked at certain height and only then remove it.

Reply:
We store at least 40 heights of proposals, a node that is 30 heights below or more would start redownloading itself already before we would remove any proposals he may need. These parameters could change, but implementing this would introduce just more complexity and with the current config would never be used. Also, another case is what if someone doesn't have proposals at all, should we keep all of them indefinitely? If so memory leak, so another case to handle.

Reply:
I get your point. If you think it's too much hassle and no gain then there is no point in implementing that indeed. Especially if the logic right now is sealed e.i. all the cases are handled and we won't have nodes continuing ahead indefinitely - in consequence the redownload will work as intended.